frontend/src/modules/entitydetails/components/EntityDetails.js

```diff
@@ -97,6 +97,20 @@ export class EntityDetailsBase extends React.Component<InternalProps, State> {
         dispatch(otherlocales.actions.get(parameters.entity, parameters.locale));
     }
 
+    customMachinerySearch = (query: string) => {
+        const { dispatch, locale, selectedEntity } = this.props;
+
+        // Deep copy entity to avoid mutating state
+        const entity = JSON.parse(JSON.stringify(selectedEntity));
+
+        // On empty query, use source string as input
+        if (entity && query.length) {
+            entity.original = query;
+        }
```

Comment:
Obviously, this is ugly. I believe the proper solution here is to refactor the machinery API:
* now: `machinery.actions.get(entity, locale)`
* after refactor: `machinery.actions.get(source, pk, locale)`

Reply:
Yes, I entirely agree. It also makes the `get` action more focused, so that's all the better.